parcels/fieldset.py

```diff
@@ -778,16 +778,23 @@ def computeTimeChunk(self, time, dt):
                 continue
             g = f.grid
             if g.update_status == 'first_updated':  # First load of data
+                if f.data is not None and not isinstance(f.data, DeferredArray):
+                    if not isinstance(f.data, list):
+                        f.data = None
+                    else:
+                        for i in range(len(f.data)):
+                            del f.data[i, :]
```

Comment:
Why does this need to be done for each `i` separately?

Reply:
that is because, if called as `del f.data[:]`, one cannot address the sub-elements anymore - they would need to be reallocated as `f.data = [None, ] * 3`. That's not necessary, because we only want to remove the larger field data within the python list, not the list itself. That is why it's written the way it is, removing each sub-element individually. It's als `range(len(f.data))` to avoid problems when only having a period of two (or one) discrete time periods. We will see if this is a performance bottleneck now, but it's a short loop and only occurs after (re-)initialization, so I don't expect a large hit on performance in common use cases.

Reply:
OK that's clear. makes sense then